apps/ogr2ogr_lib.cpp

```diff
@@ -2328,19 +2340,58 @@ GDALDatasetH GDALVectorTranslate(const char *pszDest, GDALDatasetH hDstDS,
             *pbUsageError = TRUE;
         return nullptr;
     }
+    if (psOptions->poClipSrc && !psOptions->poClipSrc->IsValid())
+    {
+        if (!psOptions->bMakeValid)
+        {
+            CPLError(CE_Failure, CPLE_IllegalArg,
+                     "-clipsrc geometry is invalid. You can try to make it "
+                     "valid with -makevalid");
```

Comment:
The results of MakeValid can be surprising.

Do you think it makes sense to either extend this message along the lines of

> -clipsrc geometry is invalid. You can try to make it valid with -makevalid. The results should be manually inspected.

or instead have a separate warning when MakeValid is actually used?

> The -clipsrc geometry is invalid and was corrected with MakeValid. The results of this operation should be inspected.

Reply:
> The results should be manually inspected.

Not sure if that would be very helpful, as the user has no easy way to know how the geometry was corrected by MakeValid()

Reply:
Maybe more clear to say "the output of ogr2ogr should be manually inspected."

Reply:
ok, warning/error messages tuned